lib/constants.ts

```diff
@@ -17,10 +17,11 @@ export const DECORATOR_FLAGS = Object.freeze({
   TaskFailureHandlerEndLabel: "_End_With_Failure",
 });
 
-export const PLAYBOOK_FORMATTER_STEP_NAME = "PLAYBOOK_FORMATTER";
-export const PLAYBOOK_DIRECT_INVOCATION_CHECK_STEP_NAME =
-  "Was_Playbook_Direct_Executed";
-export const PLAYBOOK_SETUP_STEP_NAME = "Setup_Socless_Global_State";
+// Here we explicitly assign `string` to prevent `const` type incompatibility with Record<string, State>
+export const PLAYBOOK_FORMATTER_STEP_NAME = "PLAYBOOK_FORMATTER" as string;
+export const PLAYBOOK_DIRECT_INVOCATION_CHECK_STEP_NAME = "Was_Playbook_Direct_Executed" as string;
+export const PLAYBOOK_SETUP_STEP_NAME = "Setup_Socless_Global_State" as string;
```

Comment:
i did this first, but thought the `as string` was more explicit in what we're trying to accomplish

Reply:
I think they're both equally explicit. But I think `as type` is built for a different usecase than the way its used here.

In our case here, I _think_ because we're using `as string` at the time of const assignment, the "cast a literal to a different type" never actually happens because Typescript will always treat the const as a string (it won't ever treat it as a literal type).  So given the following contrived example, Typescript will yell. 
```
 // function that expects a literal "Hello"
function greet(greet_word: "Hello"){

    console.log(greet_word)
}


const hello = "Hello" as string;  // treated as equivalent to const hello: string = "Hello"

greet(hello) // typescript yells here because hello isn't a literal-type 
```

I _think_ the `as string` usecase is for scenarios where we want to always maintain the literal type but want to cast to a different type as-needed. So something along the lines of

```
// greet function elided for brevity

const literal_hello = "Hello" // we'll always have this literal "Hello"

const string_hello = literal_hello as string // but we can make a string-type version of it here as well

greet(literal_hello) // typescript no yell
greet(string_hello) // typescript yell
greet(string_hello as "Hello") // no more typescript yelling 
```

So I'd say `export const PLAYBOOK_FORMATTER_STEP_NAME: string = "PLAYBOOK_FORMATTER"` is equally explicit and more idiomatic (provided that we don't need to export and use the literal type. If we do need to export the literal type then we might need to decouple the `as string` from the const assignment

Give it one more once over and let me know what you think